samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -37,39 +37,147 @@ static K_SEM_DEFINE(sem_stream_qos, 0, 1);
 static K_SEM_DEFINE(sem_stream_enabled, 0, 1);
 static K_SEM_DEFINE(sem_stream_started, 0, 1);
 
-void print_hex(const uint8_t *ptr, size_t len)
+
+#if defined(CONFIG_LIBLC3CODEC)
+
+#include "lc3.h"
+#include "math.h"
+
+/* Current sample do not use codec configuration parameters, hence below shall match the selected
+ * codec configuration.
+ * One sample data buffer en generated and repeated.
+ */
+#define AUDIO_SAMPLE_RATE_HZ    16000
+#define AUDIO_FREQUENCY_HZ      400
+#define AUDIO_LENGTH_US         10000 /* amount of sample data - shall match LC3 frame length */
+#define AUDIO_LENGTH_100US      (AUDIO_LENGTH_US / 100)
+#define NUM_SAMPLES             ((AUDIO_LENGTH_US * AUDIO_SAMPLE_RATE_HZ) / USEC_PER_SEC)
+#define AUDIO_VOLUME            (INT16_MAX - 3000) /* codec does clipping above INT16_MAX - 3000 */
+
+static int16_t audio_buf[NUM_SAMPLES];
+static lc3_encoder_t lc3_encoder;
+static lc3_encoder_mem_48k_t lc3_encoder_mem;
+
+
+/* Some of the prints made from this application take too long causing the applicaiton to fail.
+ * Enable only when the debug terminal can handle it - e.g. baudrate > 115200
+ */
+#define VERBOSE false
+
+/**
+ * Use the math lib to generate a sine-wave using 16 bit samples into a buffer.
+ *
+ * @param buf Destination buffer
+ * @param length_us Length of the buffer in microseconds
+ * @param frequency_hz frequency in Hz
+ * @param sample_rate_hz sample-rate in Hz.
+ */
+static void fill_audio_buf_sin(int16_t *buf, int length_us, int frequency_hz, int sample_rate_hz)
 {
-	while (len-- != 0) {
-		printk("%02x", *ptr++);
+	const int sine_period_samples = sample_rate_hz / frequency_hz;
+	const unsigned int num_samples = (length_us * sample_rate_hz) / USEC_PER_SEC;
+	const float step = 2 * 3.1415 / sine_period_samples;
+
+	for (int i = 0; i < num_samples; i++) {
+		float sample = sin(i * step);
```

Comment:
```suggestion
		const float sample = sin(i * step);
```

Reply:
Changed, but I think the compiler is able to work that out ;-)